code/modules/pool/pool_controller.dm

```diff
@@ -0,0 +1,415 @@
+#define POOL_NO_OVERDOSE_MEDICINE_MAX 5		//max units of no-overdose medicine to allow mobs to have through duplication
+
+//Originally stolen from paradise. Credits to tigercat2000.
+//Modified a lot by Kokojo and Tortellini Tony for hippiestation.
+//Heavily refactored by tgstation
+/obj/machinery/pool
+	icon = 'icons/obj/machines/pool.dmi'
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE
+
+/obj/machinery/pool/controller
+	name = "\improper Pool Controller"
+	desc = "An advanced substance generation and fluid tank management system that can refill the contents of a pool to a completely different substance in minutes."
+	icon_state = "poolc_3"
+	density = TRUE
+	use_power = TRUE
+	idle_power_usage = 75
+	/// How far it scans for pool objects
+	var/scan_range = 6
+	/// Is pool mist currently on?
+	var/mist_state = FALSE
+	/// Linked mist effects
+	var/list/obj/effect/mist/linked_mist = list()
+	/// Pool turfs
+	var/list/turf/open/pool/linked_turfs = list()
+	/// All mobs in pool
+	var/list/mob/living/mobs_in_pool = list()
+	/// Is the pool bloody?
+	var/bloody = 0
+	/// Last time we process_reagents()'d
+	var/last_reagent_process = 0
+	/// Maximum amount we will take from a beaker
+	var/max_beaker_transfer = 100
+	/// Minimum amount of a reagent for it to work on us
+	var/min_reagent_amount = 10
+	/// ADMINBUS ONLY - WHETHER OR NOT WE HAVE NOREACT ;)
+	var/noreact_reagents = FALSE
+	/// how fast in deciseconds between reagent processes
+	var/reagent_tick_interval = 5
+	/// Can we use unsafe temperatures
+	var/temperature_unlocked = FALSE
+	/// See __DEFINES/pool.dm, temperature defines
+	var/temperature = POOL_NORMAL
+	/// Whether or not the pool can be drained
+	var/drainable = FALSE
+	// Whether or not the pool is drained
+	var/drained = FALSE
+	/// Pool drain
+	var/obj/machinery/pool/drain/linked_drain
+	/// Pool filter
+	var/obj/machinery/pool/filter/linked_filter
+	/// Next world.time you can interact with settings
+	var/interact_delay = 0
+	/// Airlock style shocks
+	var/shocked = FALSE
+	/// Old reagent color, used to determine if update_color needs to reset colors.
+	var/old_rcolor
+	/// Just to prevent spam
+	var/draining = FALSE
+	/// Reagent blacklisting
+	var/respect_reagent_blacklist = TRUE
+
+/obj/machinery/pool/controller/examine(mob/user)
+	. = ..()
+	. += "<span class='boldnotice'>Alt click to drain reagents.</span>"
+
+/obj/machinery/pool/controller/Initialize()
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+	create_reagents(1000)
+	if(noreact_reagents)
+		reagents.reagents_holder_flags |= NO_REACTION
```

Comment:
```suggestion
		reagents.reagents_holder_flags |= NO_REACT
```

NO_REACTION is the a gas_mixture define. probably one, another or both should be renamed one day.